src/backend/distributed/utils/reference_table_utils.c

```diff
@@ -605,34 +624,6 @@ DeleteAllReferenceTablePlacementsFromNodeGroup(int32 groupId)
 }
 
 
-/*
- * ReferenceTableOidList function scans pg_dist_partition to create a list of all
- * reference tables. To create the list, it performs sequential scan. Since it is not
- * expected that this function will be called frequently, it is OK not to use index scan.
- * If this function becomes performance bottleneck, it is possible to modify this function
- * to perform index scan.
- */
-List *
-ReferenceTableOidList()
```

Comment:
This implementation has race conditions if caller doesn't acquire enough locks. tables can be dropped after `DistTableOidList()` returns and before we call `GetCitusTableCacheEntry()`.